Add `firstInterimResponseStart` and `deliveryType` to PerformanceResourceTiming (#28569)
mdn/content: files/en-us/web/api/performanceresourcetiming/deliverytype/index.md

---
title: "PerformanceResourceTiming: deliveryType property"
short-title: deliveryType
slug: Web/API/PerformanceResourceTiming/deliveryType
page-type: web-api-instance-property
browser-compat: api.PerformanceResourceTiming.deliveryType
---

{{APIRef("Performance API")}}

> The **`deliveryType`** read-only property is a string representing web platform feature that exposes information about how the resource was delivered. For example, resources which were delivered from the cache.

Reviewer: ```suggestion
The **`deliveryType`** read-only property is a string indicating how the resource was delivered — for example, indicating whether the resource was delivered from the cache.
```

## Value

> The `deliveryType` property can have the following values:.
> 
> - `cache`
>   - : If the request was retrieved from the cache.
> - `""` (empty string)
>   - : The empty string is returned otherwise.
> 
> This is expected to be expanded by future updates to this specification, e.g. to describe consuming preloaded resources and prefetched navigation requests.

Reviewer: ```suggestion
The `deliveryType` property can have the following values:.

- `cache`
  - : If the resource was retrieved from the cache
- `""` (empty string)
  - : If none of the other defined delivery types

This set of delivery types is expected to be expanded in the future — for example, to indicate preloaded resources and to indicate prefetched navigation requests.
```

## Examples

### Filtering resources

> The `deliveryType` property can be used to get specific resource timing entries only. For example, only those that were cached.

Reviewer: ```suggestion
The `deliveryType` property can be used to get specific resource timing entries only; for example, only those that were cached.
```

> Example using a {{domxref("PerformanceObserver")}}, which notifies of new `resource` performance entries as they are recorded in the browser's performance timeline. Use the `buffered` option to access entries from before the observer creation.

Reviewer: ```suggestion
The following example uses a {{domxref("PerformanceObserver")}} which notifies of new `resource` performance entries as they are recorded in the browser's performance timeline. The `buffered` option is used for accessing entries from before the observer creation.
```

```js
const observer = new PerformanceObserver((list) => {
  const cachedResources = list.getEntries().filter((entry) => {
    return entry.deliveryType === "cache";
  });
  console.log(cachedResources);
});

observer.observe({ type: "resource", buffered: true });
```

> Example using {{domxref("Performance.getEntriesByType()")}}, which only shows `resource` performance entries present in the browser's performance timeline at the time you call this method:

Reviewer: ```suggestion
The following example uses {{domxref("Performance.getEntriesByType()")}}, which only shows `resource` performance entries present in the browser's performance timeline at the time you call the method.
```
